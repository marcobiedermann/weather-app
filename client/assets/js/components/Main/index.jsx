import PropTypes from 'prop-types';
import React from 'react';

const Main = props => {
  const { children, ...otherProps } = props;

  return <main {...otherProps}>{children}</main>;
};

Main.propTypes = {
  children: PropTypes.node,
};

Main.defaultProps = {
  children: null,
};

export default Main;
