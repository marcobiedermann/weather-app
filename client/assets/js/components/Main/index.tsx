import React, { FC } from 'react';

const Main: FC = (props) => {
  return <main {...props} />;
};

export default Main;
