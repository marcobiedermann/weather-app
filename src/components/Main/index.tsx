import { ReactNode } from 'react';
import styles from './style.module.css';

interface MainProps {
  children: ReactNode;
}

function Main(props: MainProps): JSX.Element {
  return <main className={styles.main} {...props} />;
}

export type { MainProps };
export default Main;
