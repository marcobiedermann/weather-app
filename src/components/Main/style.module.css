.main {
  flex-grow: 1;
  max-width: 70%;
  padding: 3em;
}
