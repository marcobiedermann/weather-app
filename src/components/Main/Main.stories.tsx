import type { Meta, StoryObj } from '@storybook/react';
import Main from '.';

const meta: Meta<typeof Main> = {
  args: {
    children: 'Main',
  },
  component: Main,
};

type Story = StoryObj<typeof Main>;

const Default: Story = {};

export { Default };
export default meta;
