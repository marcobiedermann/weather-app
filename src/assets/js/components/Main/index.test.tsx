import React from 'react';
import renderer from 'react-test-renderer';
import Main from '.';

describe('components/Main', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Main>Content</Main>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
