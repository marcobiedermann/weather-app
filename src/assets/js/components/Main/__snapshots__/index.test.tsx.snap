// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Main renders correctly 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    Content
  </main>
</DocumentFragment>
`;
